Wikipedia/Code/WMFImageURLParsing.m

```diff
@@ -6,7 +7,7 @@
     dispatch_once(&onceToken, ^{
         // TODO: try to read serialized regex from disk to prevent needless pattern compilation on next app run
         NSError* patternCompilationError;
-        imageNameFromURLRegex = [NSRegularExpression regularExpressionWithPattern:@"^\\d+px-([^.]*(\\.[^.$]{4,})*\\.[^.]*).*$"
+        imageNameFromURLRegex = [NSRegularExpression regularExpressionWithPattern:@"^\\d+px-(.*)"
```

Comment:
@mdholloway @niedzielski @dbrant check out this change to image filename parsing regex & business logic (see below). might want to adopt in android app as well. This image URL was discovered on the [Swaging](https://en.wikipedia.org/wiki/Swaging) article in EN Wiki.